internal/server/config.go

```diff
@@ -56,5 +72,11 @@ func LoadConfig(configFile string) (Config, error) {
 		}
 		return Config{}, err
 	}
+
+	// keep for backward-compatibility
+	if cfg.AuthenticatedUserHeaderKey != "X-Auth-Email" {
```

Comment:
is this really necessary to manually check this? if we have defined default tag, the value will be used if we let this empty.
I think it is even not necessary to define the default value tag.

Reply:
the idea was to check `cfg.AuthenticatedUserHeaderKey` if it contains non-default value (such as user configure the header key on older version, let's say `X-Something`), then `cfg.Auth.Default.HeaderKey` will automatically be assigned the old value (to avoid `X-Something` accidentally changed to `X-Auth-Email`)

however, after giving it some thought, i think that's a good idea to let `cfg.AuthenticatedUserHeaderKey` without default tag so to compare it with empty string. do you have suggestions on make this backward-compatibility support better?